doc/source/msg.rst

```diff
@@ -8,21 +8,33 @@ Message Protocol
 Overview
 --------
 
-The `bluesky.Msg` object is a `namedtuple` subclass which has the fields
+The `RunEngine` oversees all commands of a run, intelligently rewinding and
+re-running commands during a run in order to produce a clean result. How does
+it work? On the basic level, it works by storing a list of atomistic operations
+to be run, and running them in sequence. Each of these atomistic operations are
+represented by a `bluesky.Msg` object. Thus, every plan designed for use by the
+`RunEngine` is a list of `Msg` objects. To fully capture the provenance of a
```

Comment:
I don't really like 'provenance' here, it feels off.  I have started to think of the `Msg` objects as analogs of machine instructions, python byte code, or an RPC call.